File: app/src/main/java/me/liam/support/ISupportFragment.java
package me.liam.support;

import android.os.Bundle;
import android.view.animation.Animation;

import me.liam.anim.FragmentAnimation;

public interface ISupportFragment {

    public boolean dispatcherOnBackPressed();

    public boolean onBackPressed();

    public FragmentAnimation onCreateCustomerAnimation();

    public void onEnterAnimEnd();

    public void onSupportPause();

    public void onSupportResume();

    public void onResult(int requestCode, int resultCode, Bundle data);

    public void setResult(int resultCode, Bundle data);

    public void loadRootFragment(int containerId, SupportFragment to, FragmentAnimation anim, boolean playEnterAnim, boolean addToBackStack);

    public void loadRootFragment(int containerId, SupportFragment to);

    public void loadMultipleRootFragments(int containerId, int showPosition, SupportFragment... fragments);

    public void showHideAllFragment(SupportFragment show);

    public void start(SupportFragment to);

    public void start(SupportFragment to, boolean addToBackStack);

    public void startForResult(SupportFragment to, int requestCode);

    public void startWithPop(SupportFragment to);

    public void pop();

    public void popTo(Class cls);

    public void popTo(Class cls, boolean includeTarget);

    public void popChild();

    public void popChildTo(Class cls);

    public void popChildTo(Class cls, boolean includeTarget);
}

File: core/src/main/java/me/liam/fragmentation/support/ISupportFragment.kt
package me.liam.fragmentation.support

import android.os.Bundle
import androidx.annotation.IdRes

import me.liam.fragmentation.anim.FragmentAnimation

interface ISupportFragment {

    val extraTransaction: ExtraTransaction

    fun <T : SupportFragment> findFragment(cls: Class<*>): T?

    fun <T : SupportFragment> findChildFragment(cls: Class<*>): T?

    fun dispatcherOnBackPressed(): Boolean

    fun onBackPressed(): Boolean

    fun onCreateCustomerAnimation(): FragmentAnimation?

    fun onLazyInit(savedInstanceState: Bundle?)

    fun onEnterAnimEnd()

    fun onSupportPause()

    fun onSupportResume()

    fun onSwipeDrag(beforeOne: SupportFragment, state: Int, scrollPercent: Float)

    fun onResult(requestCode: Int, resultCode: Int, data: Bundle?)

    fun onPostedData(code: Int, data: Bundle)

    fun setResult(resultCode: Int, data: Bundle)

    fun loadRootFragment(@IdRes containerId: Int, to: SupportFragment, anim: FragmentAnimation, playEnterAnim: Boolean, addToBackStack: Boolean)

    fun loadRootFragment(@IdRes containerId: Int, to: SupportFragment)

    fun loadMultipleRootFragments(@IdRes containerId: Int, showPosition: Int, vararg fragments: SupportFragment)

    fun showHideAllFragment(show: SupportFragment)

    fun start(to: SupportFragment, addToBackStack: Boolean = true)

    fun startForResult(to: SupportFragment, requestCode: Int)

    fun startWithPop(to: SupportFragment)

    fun startWithPopTo(to: SupportFragment, cls: Class<*>, includeTarget: Boolean = true)

    fun pop()

    fun popTo(cls: Class<*>, includeTarget: Boolean = true)

    fun popChild()

    fun popChildTo(cls: Class<*>, includeTarget: Boolean = true)

    fun popAllChild()
}
